build/steps-build.yml

```diff
@@ -55,16 +55,31 @@
     logProjectEvents: false
     createLogFile: false
 
-- task: VSTest@2
+- script: dotnet test $(Build.SourcesDirectory)/src/$(SolutionFileName) /p:Configuration=Release_Tests /p:CollectCoverage=true /p:IncludeTestAssembly=true /p:CoverletOutputFormat=cobertura /p:ExcludeByFile="**/*.g.cs" /p:Exclude="[*]*.Tests.*" --logger trx --no-build
   displayName: 'Run tests'
   condition: and(succeeded(), eq(variables['ApplicationConfiguration'], 'Release_Tests'))
+
+- task: PublishTestResults@2
+  displayName: Publish test results
+  condition: succeededOrFailed()
```

Comment:
I think you need to add eq(variables['ApplicationConfiguration'], 'Release_Tests')) to this condition, otherwise, it'll run all the time